src/m_dispatch_gen.c

```diff
@@ -0,0 +1,151 @@
+/*
+This program generates the file m_dispatch.h.
+
+Function pointers must be called at their exact type in Emscripten,
+including return type, which makes the number of cases explode (42
+instead of 6). This is not checked at compile time, but wrong usage
+crashes at runtime. The naming pattern is t_funMN for M t_int args
+and N t_floatarg args returning t_pd *, with t_vfunMN for the void-
+returning variants. The versions with a t_pd * are for pd_objectmaker,
+which returns the new object from the method, c.f. pd_newest support.
+For the same reason, bang_new() and similar can't be private in
+Emscripten, because their return type is different (as pd_objectmaker
+methods) vs regular bang methods, so they need special treatment in the
+dispatcher: the dispatcher checks explicitly for pd_objectmaker and
+calls the method at the correct type.
+
+This file only needs to be compiled and run if the object message
+passing changes in the future. (Unlikely.)
+To regenerate m_dispatch.h:
+    cd src
+    cc -o m_dispatch_gen m_dispatch_gen.c
+    ./m_dispatch_gen[.exe] <maxargs>
+
+NB: the <maxargs> argument should be at least MAXPDARGS+1 because the
+receiver itself is passed as an additional t_int argument!
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#define MAXMAXARGS 16
```

Comment:
this seems to be wrong:

using `./m_dispatch_gen 11`, i get:
```C
typedef void (*t_vfun110)(t_int i0, t_floatarg d0, t_floatarg d1, t_floatarg d2, t_floatarg d3, t_floatarg d4, t_floatarg d5, t_floatarg d6, t_floatarg d7, t_floatarg d8, t_floatarg d9);
typedef void (*t_vfun110)(t_int i0, t_int i1, t_int i2, t_int i3, t_int i4, t_int i5, t_int i6, t_int i7, t_int i8, t_int i9, t_int i10);
```

so either the MAXMAXARGS must be set to 10, or the typedef naming algorithm must be fixed (e.g. `t_vfun_i%d_f%d`)

Reply:
yikes, @millerpuckette caught me by surprise.

Reply:
you're of course right. maybe in 50 years we will support 11 type-checked arguments :) feel free to push a fix to develop.

Reply:
or just limit `MAXMAXARGS` to 10 :)

Reply:
actually, I'm wondering why we need to manually provide the number of arguments in the first place. I guess `m_dispatch_gen.c` could just include `m_pd.h` and directly use `MAXPDARG`  🤷 

but then again, I don't really care too much because we probably won't have to touch it again in the foreseeable future.

Reply:
This could also be done with a shell script, but why change a working system...